src/core/Database.cpp

```diff
@@ -355,3 +320,34 @@ const CompositeKey & Database::key() const
     return m_data.key;
 }
 
+QVariantMap& Database::kdfParams() {
+    return m_data.kdfParams;
+}
+
+void Database::setKdfParams(QVariantMap params) {
```

Comment:
`QMap` is a class that does [implicit data sharing](http://doc.qt.io/qt-5/implicit-sharing.html).

Is this a KeePass2C convention?